### ISSUE TITLE
Published post stay on stream of another pod accounts after deletion contacts

### ISSUE THREAD

**Issue by juliaviluhina** [diaspora/diaspora#6436]
Given
Account1 on pod1
Account2 on pod2
Accounts is linked between each other in some aspect
Public posts of both accounts are shown in streams of each accounts

Steps
1. Sign in account1
2. Add public post in account1
3. Sign in account2 - post from step2 is shown
4. Delete contacts between accounts in both accounts
5. Delete post from step2
6. Check in account2 visibility of post from step2 

Expected result
Deleted post from step 2 is not shown in both accounts

Actual result
Deleted post from step 2 stay be shown in account2

**Comment by cmrd-senya** [comment]
So the problem is that the retraction is not being sent to the user who are not in contacts. I suppose it can be solved by saving the list of people's ids to whom the original post was sent. However that won't work if the posts were fetched. Any ideas how to deal with that?

**Comment by Flaburgan** [comment]
>4. Delete contacts between accounts in both accounts

So yeah this looks like a subset of #6806, so I guess it can be closed. Waiting for @SuperTux88 to confirm that this kind of issues has been fixed by #6873

**Comment by SuperTux88** [comment]
No, this was not fixed by #6873, this needs more work, which should be easier now that the federation rewrite is merged.

**Comment by SuperTux88** [comment]
#7708 should have fixed that for at least public posts, but I'm not sure if it's still a problem for private posts (from reading the description of this PR, I don't remember all the details anymore ... I also started a bigger refactoring back then, which should further improve the situation, but I lost track because of the complexity ... but I should probably look back into it).

I'll keep this open until we know if this also works for private posts.
